freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+        html = html[:word_start] + colored_word + html[word_end:]
+    return html
+
+
+def _get_keyword_positions(textline, keywords):
+    """
+    Given a textline and a list of keywords, get a sorted list of positions of keywords found in the textline.
+    Example:
+        textline: "Hello world, this is Eric."
+        keywords: ["world", "eric"]
+        would return [(6, 11), (21, 25)]. Each tuple represents word start and end index in the given string.
+    """
+    positions_list = list()
+    for word in keywords:
+        if word in textline.lower():
+            starts = [m.start() for m in re.finditer(word, textline.lower())]
+            ends = [start+len(word) for start in starts]
+            for start, end in zip(starts, ends):
+                positions_list.append((start, end))
+                # print(word, start, end)
+    return sorted(positions_list)
+
+
+def _keep_longest_overlapping_substrings(positions):
+    """
+    Deduplicate overlapping positions by keeping the longest possible keywords.
+
+    Example 1:
+        document text contains string "email me at eric_bass@some-email.com"
+        keywords vocabulary has "eric", "bass", "eric_bass"
+        Thus keywords positions will have three tuples [(12, 16), (17, 21), (12, 21)].
+        As we cannot overlap color spans, we have to deduplicate.
+        In this examples we could either keep "eric" and "bass" or to keep "eric_bass" only and delete two others.
+        We make a choice to keep the longest possible substring, deleting every other keyword overlapping with it.
+        In this example the final output after deduplication will be [(12, 21].
+
+    Example 2:
+        document text contains string "administrative procedures"
+        keywords vocabulary has "administrative", "mini", "procedure", "pro"
+        Thus keywords positions are [(0, 14), (2, 6), (15, 24), (15, 18)].
+        Keeping longest possible keywords will eliminate "mini" and "pro" keywords,
+        which were just accidentally matched and would represent a noise if kept.
+        So, after deduplication only two tuples are left: [(0, 14), (15, 24)]
+    """
+    overlap = True
+    while overlap:
+        overlap = False
+        for idx in range(len(positions)-1):
+            current_start, current_end = positions[idx]
+            next_start, next_end = positions[idx+1]
+            if current_end > next_start:
+                overlap = True
+                # In case of overlap, keep the longer substring,
+                # and delete the indices of a shorter one from the list
+                # Then break from the for-loop and restart analyzing the list
+                if (current_end - current_start) > (next_end - next_start):
+                    del positions[idx+1]
+                else:
+                    del positions[idx]
+                break
+    return sorted(positions)
+
+
+def _wrap_in_colored_span(word, score):
+    rgba = _score_to_rgb(score, COLORMAP)

```

Comment:
`_replace_with_color_spans`,  `_wrap_in_colored_span` and  `explain_logistic_regression` should all take in the `colormap` argument, this way would not have to pass `COLORMAP` as a global variable.